productivity-suite/piercing-library/src/piercing-fragment-host/piercing-fragment-host.ts

```diff
@@ -1,108 +1,100 @@
-import type { PiercingEvent } from "../piercing-events";
-import { piercingEventType } from "../piercing-events";
+import {
+  DispatchOptions,
+  getBus,
+  MessageBus,
+  MessageHandler,
+} from "../message-bus";
 
 // Important: we only need to import the type, do not import the class itself
 //            as it would bloat the script (which needs to be very light)
 import type { PiercingFragmentOutlet } from "../piercing-fragment-outlet";
 
-export class PiercingFragmentHost extends HTMLElement {
-	fragmentId!: string;
-	piercingEventsQueue: CustomEvent<PiercingEvent>[] = [];
-	queueEventListener?: (event: Event) => void;
-	stylesEmbeddingObserver?: MutationObserver;
+export class PiercingFragmentHost extends HTMLElement implements MessageBus {
```

Comment:
@petebacondarwin instead of having a `FragmentMessageBus` field here I just figured the `PiercingFragmentHost` can implement the `MessageBus` interface itself, do you see some possible issues with it?

Reply:
How does `getBus()` find this since it is looking for elements that have the `messageBus` property? I think we are safer keeping it separate.
In the future, we could also create other kinds of MessageBus instances if we needed (not that I can think of a use-case right now)

Reply:
fixed! 🙂 

the cleaning up does seem to work nicely too! 🤩

![Screenshot 2022-10-31 at 22 23 14](https://user-images.githubusercontent.com/61631103/199121413-49b361d0-f12b-42c8-b349-1d3c0c20a6eb.png)